atspi-proxies/src/related_proxies.rs

```diff
@@ -0,0 +1,317 @@
+use crate::{
+	accessible::AccessibleProxy, action::ActionProxy, application::ApplicationProxy,
+	cache::CacheProxy, collection::CollectionProxy, component::ComponentProxy,
+	document::DocumentProxy, editable_text::EditableTextProxy, hyperlink::HyperlinkProxy,
+	hypertext::HypertextProxy, image::ImageProxy, selection::SelectionProxy, table::TableProxy,
+	table_cell::TableCellProxy, text::TextProxy, value::ValueProxy, AtspiError,
+};
+use atspi_common::{Interface, InterfaceSet, Result};
+
+/// Easily acquire the other interface proxies an object may have.
+///
+/// Equip objects with conversions to proxies of the objects' further implemented interfaces
+/// by extending `AccessibleProxy`.
+///
+/// The `proxies` method returns a `Proxies` object, which contains the related interface proxies.
+///
+/// # Lazy initialization and cheap checks
+///
+/// Proxies are lazily initialized, so they are only created when requested.
+/// Interface availability is checked before creating the proxy and is a cheap bitop.
+pub trait ObjectProxies<'a> {
```

Comment:
I'd appreciate something like `ProxyExt` or something like that. Thoughts?

Reply:
Isn't naming supposed to be the hardest thing in software engineering? 
`ProxyExt` sounds just fine.